packages/gatsby-source-wordpress/src/steps/source-nodes/fetch-nodes/fetch-referenced-media-items.js

```diff
@@ -343,7 +343,16 @@ export const fetchMediaItemsBySourceUrl = async ({
 
   // take our previously cached id's and get nodes for them
   const previouslyCachedMediaItemNodes = await Promise.all(
-    cachedMediaItemNodeIds.map(async nodeId => helpers.getNode(nodeId))
+    cachedMediaItemNodeIds.map(async nodeId => {
+      const node = await helpers.getNode(nodeId)
+
+      const parentNode =
+        node?.internal?.type === `File` && node?.parent
+          ? helpers.getNode(node.parent)
+          : null
+
+      return parentNode || node
+    })
```

Comment:
I noticed this was a separate bug where we were returning the wrong kind of node since the code surrounding this expects MediaItem nodes, not File nodes